### PR TITLE
Eval metadata batch 1: BillSum, CoNLL2003, CoNLLPP, CUAD, Emotion, GigaWord, GLUE, Hate Speech 18,  Hate Speech

### PR DESCRIPTION
**Pull request by sashavor**
Adding evaluation metadata for:
- BillSum
- CoNLL2003
- CoNLLPP
- CUAD
- Emotion
- GigaWord
- GLUE
- Hate Speech 18 
- Hate Speech Offensive

### CODE REVIEW

**Comment by sashavor** [comment]
Summary of CircleCI errors:
- BoolQ: missing 8 required positional arguments: 'annotations_creators', 'language_creators', 'licenses', 'multilinguality', 'size_categories', 'source_datasets', 'task_categories', and 'task_ids'
- Conllpp: expected some content in section `Citation Information` but it is empty.
- GLUE:   'annotations_creators', 'language_creators', 'source_datasets' :['unknown'] are not registered tags
- ConLL2003:  field 'task_ids':  ['part-of-speech-tagging'] are not registered tags for 'task_ids'
-  Hate_speech18:   Expected some content in section `Data Instances` but it is empty, Expected some content in section `Data Splits` but it is empty

File: datasets/cuad/README.md [hunk 1]
```diff
@@ -20,6 +20,19 @@ task_ids:
...
+train-eval-index:
+- config: default
+  task: question-answering
+  task_id: extractive_question_answering
+  splits:
+    train_split: train
+    eval_split: test
+  col_mapping:
+    question: question
+    answers.text: answers.text
```
Comment (lewtun) [inline datasets/cuad/README.md:32]:
```suggestion
    answers.text: answers.text
    answers.answer_start: answers.answer_start
```

File: datasets/cuad/README.md [hunk 2]
```diff
@@ -20,6 +20,19 @@ task_ids:
...
 pretty_name: CUAD
+train-eval-index:
+- config: default
+  task: question-answering
+  task_id: extractive_question_answering
+  splits:
+    train_split: train
+    eval_split: test
+  col_mapping:
+    question: question
```
Comment (lewtun) [inline datasets/cuad/README.md:31]:
```suggestion
    question: question
    context: context
```

File: datasets/gigaword/README.md [hunk 3]
```diff
@@ -19,6 +19,33 @@ task_ids:
...
+  task: text-generation
+  task_id: summarization
+  splits:
+    train_split: train
+    eval_split: test
+  col_mapping:
+    document: text
+    summary: target
+  metrics:
+    - type: rouge
```
Comment (lewtun) [inline datasets/gigaword/README.md:33]: Same comment as above about grouping all ROUGE scores in one config

**Review by lewtun — APPROVED** [review]
Thank you so much for adding the evaluation metadata to all these dataset cards @sashavor ❤️ !!

I've left a few minor questions / nits, but otherwise this is looking really good. 

Regarding the CI failures, I think this occurs because the original dataset cards are missing some additional metadata. @lhoestq should we fix this in these PRs or is it OK to merge without doing so?

File: datasets/boolq/README.md [hunk 4]
```diff
@@ -3,6 +3,55 @@ languages:
 - en
 paperswithcode_id: boolq
 pretty_name: Boolean Questions
+train-eval-index:
+- config: default
+  task: question-answering
+  task_id: extractive_question_answering
```
[the remark this answers is not in the document]
Comment (lhoestq) [reply datasets/boolq/README.md:9]: It is a yes-no QA dataset, so it is a (multi-input) classification indeed (similar to NLI for example)

File: datasets/billsum/README.md [hunk 5]
```diff
@@ -19,6 +19,33 @@ task_ids:
 - summarization-other-bills-summarization
 paperswithcode_id: billsum
 pretty_name: BillSum
+train-eval-index:
+- config: default
+  task: text-generation
```
Comment (lhoestq) [inline datasets/billsum/README.md:24]: we have "summarization" already as a task_category
```suggestion
  task: summarization
```

File: datasets/gigaword/README.md [hunk 6]
```diff
@@ -19,6 +19,33 @@ task_ids:
 - summarization--other-headline-generation
 paperswithcode_id: null
 pretty_name: Gigaword
+train-eval-index:
+- config: default
+  task: text-generation
```
Comment (lhoestq) [inline datasets/gigaword/README.md:24]: same here
```suggestion
  task: summarization
```

**Review by lhoestq — COMMENTED** [review]
Thanks !

**Comment by lhoestq** [comment]
And yes we can ignore all the CI errors related to missing content in the dataset cards, these issues can be fixed in other PRs

File: datasets/conll2003/README.md [hunk 7]
```diff
@@ -20,6 +20,19 @@ task_ids:
...
+  task: token-classification
+  task_id: entity_extraction
+  splits:
+    train_split: train
+    eval_split: test
+  col_mapping:
+    tokens: tokens
+    ner_tags: tags
+  metrics:
+    - type: coval
```
[the remark this answers is not in the document]
Comment (sashavor) [reply datasets/conll2003/README.md]: updated!

File: datasets/conllpp/README.md [hunk 8]
```diff
@@ -19,6 +19,19 @@ task_ids:
...
+  task: token-classification
+  task_id: entity_extraction
+  splits:
+    train_split: train
+    eval_split: test
+  col_mapping:
+    tokens: tokens
+    ner_tags: tags
+  metrics:
+    - type: coval
```
[the remark this answers is not in the document]
Comment (sashavor) [reply datasets/conllpp/README.md]: updated!

File: datasets/billsum/README.md [hunk 9]
```diff
@@ -19,6 +19,33 @@ task_ids:
...
+  task_id: summarization
+  splits:
+    train_split: train
+    eval_split: test
+  col_mapping:
+    text: text
+    summary: target
+  metrics:
+    - type: rouge
+      name: Rouge
```
[the remark this answers is not in the document]
Comment (sashavor) [reply datasets/billsum/README.md:34]: done!

File: datasets/boolq/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sashavor) [reply datasets/boolq/README.md:9]: so what do I put as `task_id`? `multi_label_classification`?

[the remark this answers is not in the document]
Comment (lhoestq) [reply datasets/boolq/README.md:9]: I think we can ignore it for now, until multi-input-classification is implemented

[the remark this answers is not in the document]
Comment (sashavor) [reply datasets/boolq/README.md:9]: sounds good!

**Review by lhoestq — APPROVED** [review]
Thanks !

**Pull request merged**